dockstore-integration-testing/src/test/java/io/dockstore/webservice/UserResourceIT.java

```diff
@@ -513,4 +513,54 @@ public void testUpdateUserMetadataFromGithub() {
         assertEquals("DockstoreTestUser2", userProfile.getUsername());
     }
 
+    @Test
+    public void testSetUserPrivilege() {
+        io.dockstore.openapi.client.ApiClient adminWebClient = getOpenAPIWebClient(ADMIN_USERNAME, testingPostgres);
+        io.dockstore.openapi.client.ApiClient userWebClient = getOpenAPIWebClient(USER_2_USERNAME, testingPostgres);
+
+        io.dockstore.openapi.client.model.PrivilegeRequest privilegeRequest = new PrivilegeRequest();
+        io.dockstore.openapi.client.api.UsersApi adminApi = new io.dockstore.openapi.client.api.UsersApi(adminWebClient);
+        io.dockstore.openapi.client.api.UsersApi userApi = new io.dockstore.openapi.client.api.UsersApi(userWebClient);
+        io.dockstore.openapi.client.model.User admin = adminApi.getUser();
+        io.dockstore.openapi.client.model.User user = userApi.getUser();
+
+        privilegeRequest.setAdmin(false);
+        adminApi.setUserPrivileges(privilegeRequest, user.getId());
+        adminApi.setUserPrivileges(privilegeRequest, user.getId());
+        assertFalse(userApi.getUser().isIsAdmin());
+        assertFalse(userApi.getUser().isCurator());
+
+        privilegeRequest.setCurator(true);
+        adminApi.setUserPrivileges(privilegeRequest, user.getId());
+        assertFalse(userApi.getUser().isIsAdmin());
+        assertTrue(userApi.getUser().isCurator());
+
+        try {
+            userApi.setUserPrivileges(privilegeRequest, admin.getId());
+            fail("Curator should not be able to set admin permissions");
+        } catch (ApiException ex) {
+            assertEquals(ex.getCode(), HttpStatus.SC_FORBIDDEN);
+        }
+
+        privilegeRequest.setAdmin(true);
+        adminApi.setUserPrivileges(privilegeRequest, user.getId());
+        assertTrue(userApi.getUser().isIsAdmin());
+
+        privilegeRequest.setAdmin(false);
+        try {
+            adminApi.setUserPrivileges(privilegeRequest, admin.getId());
+            fail("User should not be able to set their own permissions");
+        } catch (ApiException ex) {
+            assertEquals(ex.getCode(), HttpStatus.SC_FORBIDDEN);
+        }
+
+        privilegeRequest.setCurator(false);
+        adminApi.setUserPrivileges(privilegeRequest, user.getId());
+        try {
```

Comment:
Can you assert that the user is both not an admin nor a curator? I have to follow your code from above to figure that out; and what if that code changes? -- it could have a side effect on this test. Just to make sure you are testing with the right thing.

Reply:
Done.